packages/playwright-core/src/server/firefox/ffBrowser.ts

```diff
@@ -305,10 +305,14 @@ export class FFBrowserContext extends BrowserContext {
   }
 
   async addCookies(cookies: channels.SetNetworkCookie[]) {
-    const cc = network.rewriteCookies(cookies).map(c => ({
-      ...c,
-      expires: c.expires === -1 ? undefined : c.expires,
-    }));
+    const cc = network.rewriteCookies(cookies).map(c => {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const { _chromiumHasCrossSiteAncestor, topLevelSite, ...rest } = c;
```

Comment:
A call for isolation: pick those you need instead of filtering out implementation details of other ports you are not supposed to be aware of.